packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -1088,8 +1088,12 @@ class FocusNode with DiagnosticableTreeMixin, ChangeNotifier {
     FocusOnKeyEventCallback? onKeyEvent,
     FocusOnKeyCallback? onKey,
   }) {
+    assert(
+      onKey != null || this.onKey == null,
+      'Overwriting `focusNode.onKey` to null may not be an intentional design.'
+    );
     _context = context;
-    this.onKey = onKey ?? this.onKey;
+    this.onKey = onKey;
     this.onKeyEvent = onKeyEvent ?? this.onKeyEvent;
```

Comment:
@dkwingsmt ,Hi, I am not sure how to do with the `onKeyEvent`, what do you think?

Reply:
Behaviors of `onKeyEvent` should be parallel to those of `onKey`, so yes we should change it too.

Reply:
But I think taking `null` as rewriting to `null` is counter-intuitive. If the developer really wants to rewrite the key handler to "not do anything", they can use an empty callback `() => KeyEventResult.ignore`. Having to specify the `onKey` on every call is pretty tedious, considering it's already assignable for many times before that.

Reply:
If we eventually decided to move forward with this change, we need to update the doc too.